h/auth.py

```diff
@@ -158,9 +76,8 @@ def groupfinder(userid, request):
     return list(principals)
 
 
-def effective_principals(userid, request, groupfinder=groupfinder):
-    """
-    Return the list of effective principals for the passed userid.
+def effective_principals(userid, request, groupfinder_=groupfinder):
```

Comment:
There's now no need for separate `groupfinder` and `effective_principals` functions. You can just merge the functionality of `groupfinder` directly into `effective_principals`, and make the `groupfinder` tests into `effective_principals` tests.